File: 0036-Valid-Sudoku/0036-Valid-Sudoku/0036-Valid-Sudoku.go
func check(board [][]byte, row int, col int, target byte) bool {
    for i := 0; i < 9; i++ {
        if i != col  && board[row][i] == target {
            return true
        }
        
        if i != row && board[i][col] == target {
            return true
        }
    }

    st := (row/3)*3
    en := (col/3)*3
    for i := st; i < st + 3; i++ {
        for j := en; j < en + 3; j++ {
            if (i != row || j != col) && board[i][j] == target {
                return true
            }
        }
    }
    return false
} 
func isValidSudoku(board [][]byte) bool {
    for i := 0; i < 9; i++ {
        for j := 0; j < 9; j++ {
            if board[i][j] != '.' && check(board, i, j, board[i][j]) {
                return false
            }
        }
    }
    return true
    
}

File: 0036-Valid-Sudoku/0036-Valid-Sudoku.java
class Solution {
    static boolean check(char[][] board, int row, int col, char target) {
        int no = 0;
        for (int i = 0; i < board.length; i++) {
            if (board[row][i] == target) { no++; }
            if (board[i][col] == target) { no++; }
        }
        int r = row - (row % 3);
        int c = col - (col % 3);
        for (int i = r; i < r + 3; i++) {
            for (int j = c; j < c + 3; j++) {
                if (board[i][j] == target) { no++; }
            }
        }
        return no > 3;
    }
    public boolean isValidSudoku(char[][] board) {
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                if(board[i][j] != '.' && check(board, i, j, board[i][j])) {
                    return false;
                }
            }
        }
        return true;
        
    }
}
